mlir/unittests/IR/InterfaceTest.cpp

```diff
@@ -6,6 +6,8 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
+#include "mlir/Dialect/Bufferization/IR/BufferizationTypeInterfaces.h"
```

Comment:
note: i haven't seen bufferization-specific unit tests, so ended up adding all tests here. these seem to test "TestDialect".

Reply:
As suggested by the path, this tests libIR capabilities. We rarely use this style of unit tests. The way to test this would be to write a test-only pass that, e.g., looks at function signatures and adds an attribute to the function indicating whether the types of the function results implement certain interface. It can then live in `mlir/test/lib/Dialect/Bufferization`. There is already a precedent.

Reply:
thanks for the pointer! i'll add it a bit later (seems to be quite a endeavour).

Reply:
this is now re-done according to the suggestion. thanks!